### PR TITLE
Make ZkBookieRackAffinityMapping work as expected

### PR DESCRIPTION
**Pull request by sijie**
### Motivation

The current bookie rack affinity logic is problematic. The rack map is expecting a `host:port` pair but the bookkeeper DNS resolver is expecting a hostname or a host address. 

### Modification

- Introduce a new HashMap to keep a mapping between hostname and bookie info. It maintains the mapping for bookkeeper DNS resolver to lookup the network location.

### Verification

I have verified it worked well in setting up rackware placement policy.

### CODE REVIEW

**Comment by jiazhai** [comment]
@sijie Seems the test failure is related with this change:
https://github.com/apache/pulsar/pull/6917/checks?check_run_id=662411026

```
[ERROR] Tests run: 6, Failures: 3, Errors: 0, Skipped: 3, Time elapsed: 5.004 s <<< FAILURE! - in org.apache.pulsar.zookeeper.ZkBookieRackAffinityMappingTest
[ERROR] testBasic(org.apache.pulsar.zookeeper.ZkBookieRackAffinityMappingTest)  Time elapsed: 0.028 s  <<< FAILURE!
java.lang.RuntimeException: java.net.UnknownHostException: 127.0.0.1
	at org.apache.pulsar.zookeeper.ZkBookieRackAffinityMapping.lambda$1(ZkBookieRackAffinityMapping.java:103)
	at java.util.TreeMap.forEach(TreeMap.java:1005)
	at org.apache.pulsar.zookeeper.ZkBookieRackAffinityMapping.lambda$0(ZkBookieRackAffinityMapping.java:85)
	at java.util.TreeMap.forEach(TreeMap.java:1005)
	at org.apache.pulsar.zookeeper.ZkBookieRackAffinityMapping.updateRacksWithHost(ZkBookieRackAffinityMapping.java:84)
	at org.apache.pulsar.zookeeper.ZkBookieRackAffinityMapping.getRack(ZkBookieRackAffinityMapping.java:189)
	at org.apache.pulsar.zookeeper.ZkBookieRackAffinityMapping.resolve(ZkBookieRackAffinityMapping.java:165)
	at org.apache.pulsar.zookeeper.ZkBookieRackAffinityMappingTest.testBasic(ZkBookieRackAffinityMappingTest.java:83)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.UnknownHostException: 127.0.0.1
	at org.apache.bookkeeper.net.BookieSocketAddress.<init>(BookieSocketAddress.java:68)
```

File: pulsar-zookeeper-utils/src/main/java/org/apache/pulsar/zookeeper/ZkBookieRackAffinityMapping.java [hunk 1]
```diff
@@ -162,10 +180,19 @@ private String getRack(String bookieAddress) {
             throw new RuntimeException(e);
         }
 
+        BookieInfo bi = bookieInfoMap.get(bookieAddress);
+        if (bi == null) {
+            Optional<BookieInfo> biOpt = racksWithHost.getBookie(bookieAddress);
+            if (biOpt.isPresent()) {
+                bi = biOpt.get();
+            } else {
+                updateRacksWithHost(racksWithHost);
```
Comment (jiazhai) [inline pulsar-zookeeper-utils/src/main/java/org/apache/pulsar/zookeeper/ZkBookieRackAffinityMapping.java:189]: Do we need to update it here? seems this is a  get operation, If it is not present, there is no need to update?

**Comment by jiazhai** [comment]
would like to remove it out of 2.5.2

**Comment by sijie** [comment]
@jiazhai: fix the comment. the BookiesRackConfiguration expects the key to be `hostname:port`. We used `bookieInfoMap` for resolving the rack information.

**Pull request merged**